client/state/themes/actions/activate-theme.js

```diff
@@ -57,18 +52,7 @@ export function activateTheme(
 			} )
 			.then( async ( theme ) => {
 				if ( styleVariationSlug ) {
-					const themeStylesheet = theme.stylesheet || themeId;
-					const variations = await dispatch( getGlobalStylesVariations( siteId, themeStylesheet ) );
-					const currentVariation = variations.find(
-						( variation ) =>
-							variation.title &&
-							variation.title.split( ' ' ).join( '-' ).toLowerCase() === styleVariationSlug
-					);
-
-					if ( currentVariation ) {
-						const globalStylesId = await dispatch( getGlobalStylesId( siteId, themeStylesheet ) );
-						await dispatch( updateGlobalStyles( siteId, globalStylesId, currentVariation ) );
-					}
+					await dispatch( activateStyleVariation( themeId, siteId, styleVariationSlug ) );
```

Comment:
For theme activation, we already support passing style variation slug to the endpoint. See: https://github.com/Automattic/wp-calypso/pull/72035. This way, we only need 1 endpoint call, as opposed to 3 in this PR.

Unfortunately, it only works on wpcom. We might need to add the support in Jetpack if we want.